File: lib/adapters/mock/greeting_store.go
package mock

import (
	"context"

	"github.com/maybecake/stacks/lib/domain"
)

// MockGreetingStore is a no-op implementation of domain.GreetingStore.
// It satisfies the interface without writing to any external system.
type MockGreetingStore struct{}

var _ domain.GreetingStore = (*MockGreetingStore)(nil)

func (m *MockGreetingStore) RecordGreeting(_ context.Context, _ string, _ string) error {
	return nil
}

func (m *MockGreetingStore) GetStats(_ context.Context) ([]domain.GreetingStats, error) {
	return []domain.GreetingStats{}, nil
}

func (m *MockGreetingStore) GetNameFrequencies(_ context.Context) ([]domain.NameFrequency, error) {
	return []domain.NameFrequency{}, nil
}

File: lib/adapters/mock/greeting_store.ts
import { GreetingStats, GreetingStore, GreetingTypeStat, NameFrequency } from "../../domain/greeting";

export class MockGreetingStore implements GreetingStore {
  async recordGreeting(_type: string, _name: string): Promise<void> {
    // No-op mock — no persistence until the Postgres adapter is wired
  }

  async getStats(): Promise<GreetingStats[]> {
    return [];
  }

  async getNameFrequencies(): Promise<NameFrequency[]> {
    return [];
  }

  async listGreetingTypeStats(
    _limit: number,
    _cursor: string,
  ): Promise<{ items: GreetingTypeStat[]; nextCursor: string }> {
    return { items: [], nextCursor: "" };
  }

  async listGreetedNames(
    _limit: number,
    _cursor: string,
  ): Promise<{ items: NameFrequency[]; nextCursor: string }> {
    return { items: [], nextCursor: "" };
  }
}
